prelude.links

```diff
@@ -463,19 +463,75 @@ fun asList(t) server {
 # @param {[(String, String)]} attributes List of attribute-value-pairs
 # @param {Xml} children Child XML nodes
 # @returns {Xml}
-sig makeXml : (String, [(String, String)], Xml) ~> Xml
-fun makeXml (name, attributes, children) {
-  var attributes = fold_left(fun (prev, curr) {
-    var (attr, value) = curr;
-    prev ^^ " " ^^ attr ^^ "=\"" ^^ value ^^ "\""
-  }, "", attributes);
+sig makeXml : (String, [(String, String)], Xml) ~> XmlItem
+fun makeXml (name, attrs, children) {
+  hd(variantToXml([ Node(name, map(fun (attr) { var (n, v) = attr; Attr(n, v) }, attrs) ++ xmlToVariant(children)) ]))
+}
+
+# Appends a XML forest to an XmlItems children
+# @param {XmlItem} item
+# @param {Xml} child
+# @returns {XmlItem}
+sig appendChild : (XmlItem, Xml) ~> XmlItem
```

Comment:
parenthetical note: if we defined XMLItem as the variant type then we could avoid the dynamic check by constraining the first argument type to be `[|Node:(String,Xml)|]`...